src/metabase/pulse.clj

```diff
@@ -80,29 +83,73 @@
     (compare (:row dashcard-1) (:row dashcard-2))
     (compare (:col dashcard-1) (:col dashcard-2))))
 
-(defn- virtual-card-of-type?
-  "Check if dashcard is a virtual with type `ttype`.
-
-  There are currently 3 types of virtual card: text, action, link."
-  [dashcard ttype]
-  (and (map? dashcard)
-       (= ttype (get-in dashcard [:visualization_settings :virtual_card :display]))))
+(defn- link-card->url
+  [{:keys [entity url] :as link-card}]
+  (if url
+    (:url link-card)
+    (let [{:keys [db_id id model]} entity]
+      (case model
+        "card"       (urls/card-url id)
+        "dataset"    (urls/card-url id)
+        "collection" (urls/collection-url id)
+        "dashboard"  (urls/dashboard-url id)
+        "database"   (urls/database-url id)
+        "table"      (urls/table-url db_id id)))))
+
+(defn- link-card->content
+  [{:keys [entity url] :as _link-card}]
+  (if url
+    url
+    (:name entity)))
+
+(defn- link-card->description
+  [{:keys [entity url] :as _link-card}]
+  (if url
+    nil
+    (:description entity)))
+
+(defn- link-card->text
+  [link-card]
+  {:text (str (format
+                "### [%s](%s)"
+                (link-card->content link-card)
+                (link-card->url link-card))
+              (when-let [description (link-card->description link-card)]
+                (format "\n%s" description)))})
+
+(defn- dashcard-link-card->content
+  "Convert a dashcard that is a linkcard to pulse content."
+  [{viz-settings :visualization_settings :as _dashcard}]
+  (let [link-card (:link viz-settings)]
+    (cond
+      (some? (:url link-card))
+      (link-card->text link-card)
+      ;; if link card link to an entity, update the settins because
+      ;; its stored info might be out-of-date
+      (some? (:entity link-card))
+      (let [{:keys [model id]} (:entity link-card)
+            instance           (t2/select-one
+                                 (serdes.util/link-card-model->toucan-model model)
+                                 (dashboard-card/link-card-info-query-for-model model id))]
+        (when (mi/can-read? instance)
+          (link-card->text (assoc link-card :entity instance)))))))
 
 (defn- dashcard->content
```

Comment:
I thought about it but I think cond is okay here because the dispatch for this is not straightforward.

Especially treating `everything else` as `text-card` is a special case, and writing that with `:default` feels weird.